### PR TITLE
Optimizing ReliefF.fit()

### PR DESCRIPTION
**Pull request by GrantRVD**
This PR solves a few problems left over after my last changes. In particular, there are two major changes that result in significant speed-up on my machine, which will hopefully translate to others, along with a few minor changes.

### Taking KDTree.query() out of the loop
After a bit of research, I found that `sklearn.neighbors.KDTree.query` will take an entire numpy array and will split it into queries along the first axis (i.e. row by row). This allows all the nearest neighbors for each point to be obtained simultaneously with one function call. The indices for each sample can be iterated over from there.

I also successfully vectorized the entire function, to the point that the `fit` function had no loops and worked only with matrix operations. However, because this required a lot of repeating and expanding of already large arrays, the benefit of vectorization was outweighed by the extra operations needed to make it work. Thus, I opted to re-instate the loop over each sample to compute score updates, which is roughly half a second faster per call to `fit` on my machine, and the code is also much easier to read in this format - the vectorization required some obscure usage of `np.tile` and `np.reshape` that wouldn't have been pleasant to maintain or debug.

### Computing with integers instead of floats
One of the reasons the previous vectorization didn't significantly improve speed is probably because of all the floating point arithmetic required in the vectorized approach. Because the scores seem to be purely the results of increments and decrements based on features and labels, I changed `feature_scores` to initially be an integer array and performed all subsequent operations as such, then `feature_scores` is cast to an array of floating point numbers after everything is said and done. Using integers instead of floats makes the arithmetic, comparisons, dot product, and `np.argsort` calls faster without any loss of accuracy.

### Minor changes
It turns out that `KDTree.query` also has a `return_distance` keyword argument that was useful here, so the function can be made to return indices without the need to throw away distances.

As previously mentioned, `feature_scores` now initializes to an array of zeros of type `np.int64` and cast to `np.float64` at the end of the function.

Lastly, with all the other simplifications, I streamlined the names for the looping variables, since there are now only three lines of code to the loop. Using `enumerate` made it simple to pair the sample indices with their corresponding k nearest neighbors' indices returned by `KDTree.query`.

If there's interest in testing the fully vectorized version I had, I'm happy to submit it on another branch, but I think further tests will find the same slow-down that I did. If someone can think of a way to accomplish vectorization without using `np.tile`, there may still be hope for additional optimization.

### Tests
I also took the two large testing arrays in `tests.py` and saved them to a compressed numpy array file in the data directory. This makes the test file a bit cleaner and also makes it easier to update the arrays for testing if there are changes or additions. The lines added to `tests.py` demonstrate how to efficiently load the values from the `.npz` file, without needing any additional modules.

### CODE REVIEW

File: ReliefF/_version.py [hunk 1]
```diff
@@ -19,4 +19,4 @@
 SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 """
 
-__version__ = '0.1.2'
+__version__ = '0.1.3'
```
Comment (rhiever) [inline ReliefF/_version.py]: FYI: No need to update the version with each PR. I'll update the version as I send out releases. :-)

Comment (GrantRVD) [inline ReliefF/_version.py]: Got it. I'll leave `_version.py` unmodified in any future commits.

**Comment by rhiever** [comment]
Very strange... not seeing a speedup on my internal data sets again. I'm wondering if I'm doing something wrong with the testing. How are you testing the running time?

**Comment by GrantRVD** [comment]
That's odd. I'm using the IPython %timeit magic, which typically uses Python's `time` module if memory serves. I ran it for 100 loops each time, best of 3, and the version in this PR definitely outperforms the last PR I submitted, Are you sure your test code is using the modified implementation and not some other installation of ReliefF? If you're using a Jupyter notebook, hitting Shift+tab three times on the `fit` function should call up the source code that it's using, though I don't know if that's definitively the code being executed. I'm not an expert on code timing either, so it's entirely possible my speedup is coming from somewhere else.

**Comment by GrantRVD** [comment]
I've made changes to `test.py` based on previous feedback. I added the `time_fit` function so everyone can test the timing of the fit function the same way. Calling `python -m timeit "import tests; tests.time_fit()"` from the command line should print out three time values it takes to execute `fit` ten times. It takes about 3.5 seconds per loop on my machine, so about ten seconds overall. `tests.py` now imports the `timeit` module to do this.

In addition, I put the repeated data-loading code into a function, so the other test functions start by calling `get_testing_data` instead of having the same code repeated. Furthermore, I loaded the data using just `numpy`, so `pandas`, which isn't listed in the requirements, is no longer needed to run the test functions.

**Comment by rhiever** [comment]
Good idea. Will take a look at this soon -- drowning in paper deadlines. :-(

**Comment by GrantRVD** [comment]
No problem. I'm taking a break from this for the moment, so take your time. Coveralls doesn't seem to like the fact that the `time_fit` function prints things out instead of asserting, or that I have an unused option to return the csv header labels in the data-loading function. Happy to amend the previous commit if you'd like either of those fixed.

**Comment by rhiever** [comment]
I think it's because the functions aren't named `test_*()`, so it thinks that they are core functions that should be tested. I wonder if there's a way to decorate functions so nosetests/coveralls knows to ignore them.

**Comment by rhiever** [comment]
Excellent! FYI: We're discussing in-lab how we're going to expand ReliefF to support continuous features as well as continuous endpoints. We also need to implement the numerous upgrades to ReliefF that have been developed in the lab over the years. Should have more info tomorrow.
